eth/vm/base.py

```diff
@@ -440,12 +473,15 @@ def apply_all_transactions(self, transactions, base_header):
             receipts.append(receipt)
             computations.append(computation)
 
-        return result_header, receipts, computations
+        receipts_tuple = tuple(receipts)
```

Comment:
You can change `receipts` and `computations` to be tuples and then concat and reassign them in the loop. Something like:

```python
receipts = receipts + (receipt,)
```

Then you do not have to start with lists and convert them to tuples but simply build up the tuple you want directly.

Reply:
I tried that approach as follows in the beginning for which I got an error,
`receipts += (receipt)`.
But this method worked and optimized too. Glad to make both `type hinting` and `optimization` happen at the same time.